lib/Interpreter/Compatibility.h

```diff
@@ -211,15 +211,24 @@ getSymbolAddress(clang::Interpreter& I, llvm::StringRef IRName) {
 inline llvm::Expected<llvm::JITTargetAddress>
 getSymbolAddress(clang::Interpreter& I, clang::GlobalDecl GD) {
   std::string MangledName;
```

Comment:
We can't rely on `__APPLE__` here since we might be running on OSX but using remote execution and the underlying platform to be different.

How about something like:

```cpp
auto &DL = getExecutionEngine()->getDataLayout();
char GlobalPrefix = DL.getGlobalPrefix();
bool HasGlobalPrefix = (GlobalPrefix != '\0');
if (HasGlobalPrefix && (*Name).front() != GlobalPrefix)
  // Complain?
std::string Tmp((*MangledName).data() + HasGlobalPrefix,
                    (*MangledName).size() - HasGlobalPrefix);
getSymbolAddress(Tmp);
```

Reply:
`MangledName` should be behavior agnostic, if it has the prefix then we cannot guess if it's actually the name of the symbol or if it was intended for prefix i.e. it creates ambiguity. Which is why I propose that we handle the prefix on API's end. Would that be okay?